scripts/pkg/scripts/post_install.sh

```diff
@@ -0,0 +1,116 @@
+  1|2)
+    if ! getent group "<%= group %>" >/dev/null; then
+      groupadd -r "<%= group %>"
+    fi
+
+    if ! getent passwd "<%= user %>" >/dev/null; then
+      useradd -r -g "<%= group %>" -M -s /sbin/nologin \
+      -c "service user" "<%= user %>"
+    fi
+  ;;
+
+  *)
+      echo "post install script called with unknown argument \`$1'" >&2
+      exit 1
+  ;;
+esac
+
+chown <%= user %>:<%= group %> <%= dataDir %>
+chown <%= user %>:<%= group %> <%= pluginsDir %>
+
+############ Additional System Settings ############
+OPENSEARCH_HOME=<%= homeDir %>
+OPENSEARCH_DATA_DIR=<%= dataDir %>
+OPENSEARCH_CONFIG_DIR=<%= configDir %>
+OPENSEARCH_LOG_DIR=<%= logDir %>
+PRODUCT=<%= product %>
+
+mkdir -p $OPENSEARCH_HOME/data
+mkdir -p $OPENSEARCH_CONFIG_DIR
+mkdir -p /var/run/$PRODUCT
+mkdir -p $OPENSEARCH_LOG_DIR
+
+############ Additional Plugins Settings ############
+
+function pluginSettings() {
+    echo Apply Security Settings
+    sh $OPENSEARCH_HOME/plugins/opensearch-security/tools/install_demo_configuration.sh -y -i -s
+
+    echo Apply PerformanceAnalyzer Settings
+    echo 'true' > "$OPENSEARCH_HOME"/data/rca_enabled.conf
+    echo 'true' > $OPENSEARCH_CONFIG_DIR/performance_analyzer_enabled.conf
+    echo 'true' > $OPENSEARCH_CONFIG_DIR/rca_enabled.conf
+    chown -R <%= user %>:<%= group %> $OPENSEARCH_HOME
+    chown -R <%= user %>:<%= group %> $OPENSEARCH_CONFIG_DIR
+    chmod a+rw /tmp
+    
+    if ! grep -q '## OpenSearch Performance Analyzer' $OPENSEARCH_CONFIG_DIR/jvm.options; then
+       echo Add Performance Analyzer settings in $OPENSEARCH_CONFIG_DIR/jvm.options
+       CLK_TCK=`/usr/bin/getconf CLK_TCK`
+       echo >> $OPENSEARCH_CONFIG_DIR/jvm.options
+       echo '## OpenSearch Performance Analyzer' >> $OPENSEARCH_CONFIG_DIR/jvm.options
+       echo "-Dclk.tck=$CLK_TCK" >> $OPENSEARCH_CONFIG_DIR/jvm.options
+       echo "-Djdk.attach.allowAttachSelf=true" >> $OPENSEARCH_CONFIG_DIR/jvm.options
+       echo "-Djava.security.policy=file:///usr/share/opensearch/plugins/opensearch-performance-analyzer/pa_config/opensearch_security.policy" >> $OPENSEARCH_CONFIG_DIR/jvm.options
+    fi
+
+    if command -v systemctl > /dev/null; then
+        echo '# Enabling OpenSearch performance analyzer to start and stop along with opensearch.service'
+        systemctl daemon-reload
+        #systemctl enable opensearch-performance-analyzer.service || true
+    fi
+
+}
+
+if [ "$PRODUCT" = "opensearch" ]
+then
+    echo Product is OpenSearch, apply additional settings
+
+    # Redirect Logs
+    echo "path.logs: $OPENSEARCH_LOG_DIR" >> $OPENSEARCH_CONFIG_DIR/opensearch.yml
+
+    # Plugin Settings
+    pluginSettings || echo plugin settings failed
+elif [ "$PRODUCT" = "opensearch-dashboards" ]
+then
+    echo Product is OpenSearch-Dashboards, apply additional settings
+
+    # Redirect Logs
+    echo "logging.dest: $OPENSEARCH_LOG_DIR/opensearch-dashboards.log" >> $OPENSEARCH_CONFIG_DIR/opensearch_dashboards.yml
+
+fi
```

Comment:
I think the `if` is a 1-liner

```
echo "path.logs: $OPENSEARCH_LOG_DIR" >> $OPENSEARCH_CONFIG_DIR/$PRODUCT.yml
```

Generally these scripts shouldn't know the list of products they support if they are going to be generic. Otherwise split them.

Reply:
It has to know because the commands are different between OS and OSD.

Reply:
There are many different things in this script between OpenSearch and OpenSearch Dashboards (this, performance analyzer, etc.). It should be two separate scripts IMO.

Reply:
It would add more complexity to the fpm_builder.py to separate between OS and OSD.
The approach now is not ideal but solved the issues for the 1st draft.
I think we can mark this as a later improvement.

Thanks.